src/main/java/org/broadinstitute/hellbender/utils/Nucleotide.java

```diff
@@ -64,58 +65,87 @@
     public static final Nucleotide AMINO = M;
     public static final Nucleotide KETO = K;
     public static final Nucleotide ANY = N;
-    public static final Nucleotide UNKNOWN = N;
     public static final Nucleotide INVALID = X;
 
     /**
      * List of the standard (non-redundant) nucleotide values in their preferred alphabetical order.
      */
-    public static final List<Nucleotide> STANDARD_DNA_BASES = Collections.unmodifiableList(Arrays.asList(A, C, G, T));
+    public static final List<Nucleotide> STANDARD_BASES = Collections.unmodifiableList(Arrays.asList(A, C, G, T));
 
-    // actually calling values() is costly (creates a new array every time) and often we do just to find out the
-    // total number of constants.
+    // Since calling values() is costly (creates a new array every time) and often we do it just to find out the
+    // total number of constants is best to cache it in a constant.
     private static final int NUMBER_OF_CONSTANTS;
 
     private static final Nucleotide[] baseToValue;
     private static final Nucleotide[] maskToValue;
+    private static final int[] baseToOrdinal;
 
     static {
         final Nucleotide[] values = values();
         NUMBER_OF_CONSTANTS = values.length;
         baseToValue = new Nucleotide[1 << Byte.SIZE];
-        maskToValue = new Nucleotide[1 << 4];
+        maskToValue = new Nucleotide[1 << STANDARD_BASES.size()];
+        baseToOrdinal = new int[1 << Byte.SIZE];
         Arrays.fill(baseToValue, INVALID);
+        Arrays.fill(baseToOrdinal, INVALID.ordinal());
         for (final Nucleotide nucleotide : values) {
-            baseToValue[nucleotide.lowerCaseByteEncoding & 0xFF]
-                    = baseToValue[nucleotide.upperCaseByteEncoding & 0xFF] = nucleotide;
-            maskToValue[nucleotide.mask] = nucleotide;
+            final int lowerCaseIndex = nucleotide.lowerCaseByteEncoding & 0xFF;
+            final int upperCaseIndex = nucleotide.upperCaseByteEncoding & 0xFF;
+            maskToValue[nucleotide.mask]
+                    = baseToValue[lowerCaseIndex]
```

Comment:
Not sure if this adheres to Java style guide.

Reply:
I quick google does not show anything against multiple assigns, perhaps you can post a link to it.

I will break into two lines the assignations to maskToValue from the other two to baseToValue as they go into different arrays, is perhaps more clear that way.